### DISCUSSION TITLE
Converting from AnnData to Seurat via h5Seurat. Help!

### DISCUSSION THREAD

**Discussion by Matteo-Storti** [satijalab/seurat#8091] [Q&A] [answered]
Hi! I have some issues with in converting .h5ad file in the h5seurat one. I downloaded a dataset from cellxgene (https://cellxgene.cziscience.com/collections/6f6d381a-7701-4781-935c-db10d30de293) but I have errors when I load the converted seurat object. I used this code:
`#Convert h5ad format to seurat
Convert("e4eeabe8-9852-48ef-ab5b-568c64634766.h5ad", dest = "h5seurat", overwrite = TRUE)`
`Warning message:
“Unknown file type: h5ad”
Warning message:
“'assay' not set, setting to 'RNA'”
Creating h5Seurat file for version 3.1.5.9900

Adding X as data

Adding raw/X as counts

Adding meta.features from raw/var

Adding X_scanvi_emb as cell embeddings for scanvi_emb

Adding X_umap as cell embeddings for umap

Adding batch_condition to miscellaneous data

Adding default_embedding to miscellaneous data

Adding schema_version to miscellaneous data

Adding title to miscellaneous data

Adding layer soupX as data in assay soupX

Adding layer soupX as counts in assay soupX

lung <- LoadH5Seurat("e4eeabe8-9852-48ef-ab5b-568c64634766.h5seurat", misc=F)
lung
Validating h5Seurat file

Initializing RNA with data

Warning message in sparseMatrix(i = x[["indices"]][] + 1, p = x[["indptr"]][], x = x[["data"]][], :
“NAs introduced by coercion to integer range”
Error in sparseMatrix(i = x[["indices"]][] + 1, p = x[["indptr"]][], x = x[["data"]][], : 'p' must be a nondecreasing vector c(0, ...)
Traceback:

1. LoadH5Seurat("e4eeabe8-9852-48ef-ab5b-568c64634766.h5seurat", 
 .     misc = F)
2. LoadH5Seurat.character("e4eeabe8-9852-48ef-ab5b-568c64634766.h5seurat", 
 .     misc = F)
3. LoadH5Seurat(file = hfile, assays = assays, reductions = reductions, 
 .     graphs = graphs, neighbors = neighbors, images = images, 
 .     meta.data = meta.data, commands = commands, misc = misc, 
 .     tools = tools, verbose = verbose, ...)
4. LoadH5Seurat.h5Seurat(file = hfile, assays = assays, reductions = reductions, 
 .     graphs = graphs, neighbors = neighbors, images = images, 
 .     meta.data = meta.data, commands = commands, misc = misc, 
 .     tools = tools, verbose = verbose, ...)
5. as.Seurat(x = file, assays = assays, reductions = reductions, 
 .     graphs = graphs, neighbors = neighbors, images = images, 
 .     meta.data = meta.data, commands = commands, misc = misc, 
 .     tools = tools, verbose = verbose, ...)
6. as.Seurat.h5Seurat(x = file, assays = assays, reductions = reductions, 
 .     graphs = graphs, neighbors = neighbors, images = images, 
 .     meta.data = meta.data, commands = commands, misc = misc, 
 .     tools = tools, verbose = verbose, ...)
7. AssembleAssay(assay = assay, file = x, slots = assays[[assay]], 
 .     verbose = verbose)
8. as.matrix(x = assay.group[["data"]])
9. as.matrix.H5Group(x = assay.group[["data"]])
10. as.sparse(x = x, ...)
11. as.sparse.H5Group(x = x, ...)
12. sparseMatrix(i = x[["indices"]][] + 1, p = x[["indptr"]][], x = x[["data"]][], 
  .     dims = h5attr(x = x, which = "dims"))
13. stop("'p' must be a nondecreasing vector c(0, ...)")

It created the .h5seurat file but I am not able to load in the environment.
Do you have any suggestions?

**Comment by uqnsarke** [comment]
following

**Comment by denvercal1234GitHub** [comment]
Check out https://github.com/denvercal1234GitHub/anndata_Seurat_Utilities and see if it helps your specific case?

**Comment by samuel-marsh** [comment, marked as answer]
Th scverse team has released an official R package to perform such conversions.
https://github.com/scverse/anndataR
